code/modules/lighting/lighting_system.dm

```diff
@@ -72,7 +73,9 @@
 		return
 
 	radius = max(LIGHTING_MIN_RADIUS, new_luminosity)
+	radius = owner.get_light_range(radius)
```

Comment:
This proc isn't the killer. The real problems are light_source/add_effect(), light_source/remove_effect(), turf/update_lumcount(), and turf/redraw_lighting().